Automattic/woocommerce-payments
# Add Stripe Link support for 3DS payments

Fixes https://github.com/Automattic/woocommerce-payments/issues/8022

#### Changes proposed in this Pull Request
This pull request updates the usage of Stripe JS API functions. It replaces [the old `confirmCardPayment`](https://stripe.com/docs/js/payment_intents/confirm_card_payment) function with [the new `handleNextAction`](https://stripe.com/docs/js/payment_intents/handle_next_action) function. Both functions achieve the same thing, but the difference is that `confirmCardPayment` only handles next actions for card payment method while our system should support handling next actions for the Stripe Link payment method.

While working on this issue, I've noticed that the payment method type on order received page is set incorrectly when using Stripe Link with cards to perform transactions that create setup intents (e.g. subscriptions with free trials). Since after this PR we start supporting Stripe Link with 3DS cards, this misbehavior extends to 3DS cards as well. I've created https://github.com/Automattic/woocommerce-payments/issues/8122









#### Testing instructions
> Before starting, turn on Stripe Link payment method under `Payments -> Settings`

##### Test setup intent confirmation
* add a subscription with a free trial to the cart
* navigate to the checkout, ensure you're using the email address which isn't registered in Stripe Link
* initialize Stripe Link by clicking the button, insert 3DS test card details into the Payment Element, check `Save to Stripe Link` checkbox
* confirm that upon placing an order, the 3DS confirmation modal appears as expected and allows you to successfully authorize the card and complete the payment
* navigate to WooCommerce order details, confirm that the payment method is set to Stripe Link
* repeat all the steps, however when checkout page opens you should have a Stripe Link registered email address in the email field. Use it and thus test 3DS card payments with Link for a registered Link account

##### Test payment intent confirmation
* repeat the same steps as in setup intent confirmation, but replace a free-trial subscription with a physical product
* after checking all the steps from the setup intent tests, navigate to Stripe Dashboard, confirm that `POST /v1/payment_intents/{your_payment_intent_id}/confirm` request returns `HTTP 200`

-------------------


**Post merge**
